Replace references about Windows

ehmicky opened the pull request:
This library now supports some Windows terminals providing they can display Unicode symbols.
This PR replaces all references to "Windows" to "fallback", "non-Unicode" or "older terminals" instead. This includes the named export `windowsSymbols` renamed to `fallbackSymbols`.
It also replaces "OS" by "terminal".

ehmicky commented on readme.md:
```diff
@@ -1,12 +1,12 @@
 [*and more...*](index.js)
 
-Windows CMD only supports a [limited character set](http://en.wikipedia.org/wiki/Code_page_437).
+Terminals such as Windows Console Host only support a [limited character set](http://en.wikipedia.org/wiki/Code_page_437).
```
I might be wrong, but I think "Windows Console Host" (the terminal) is responsible for the poor font, while "CMD" (the shell) is just a program running on that terminal. For example, Powershell can also use Windows Console Host and exhibit the same problem.

sindresorhus replied:
Sure, but CMD cannot run anything else, so I think it would be useful to at least mention CMD in parens.

ehmicky replied:
Fixed :+1:

sindresorhus commented:
Are we ready for a release after this PR?

sindresorhus commented:
Can you fix the merge conflict?

ehmicky commented:
Conflict merged :+1: 
Yes, we can release! :tada: 
I can write down the release notes, if you want

sindresorhus commented:
I wrote some release notes, but feel free to tweak it: https://github.com/sindresorhus/figures/releases/tag/v4.0.0 (you have access)

ehmicky commented:
Awesome! :tada: :tada: :tada: 
This was a big release.
I made a few tweaks to the release notes: fixed typo (`windowsSymbols` -> `fallbackSymbols`) and added some bullet points for performance and documentation improvements.

sindresorhus commented:
Thanks for all your hard work on this 🙏
